### ISSUE TITLE
Transcript Support Level Filter error: TypeError: unsupported object type: <class 'NoneType'>

### ISSUE THREAD

**Issue by qxiong2** [griffithlab/pVACtools#287]
Hi I ran the following command and got errors, does anybody know what might be wrong?
```
pvacseq run $root_dir/pvacseq_example_data/input.vcf Test HLA-A*02:01,HLA-B*35:01,DRB1*11:01 NetMHCIIpan NetMHCpan $root_dir/results -e 8,9,10,11
/home/xx/anaconda3/lib/python3.6/site-packages/h5py/__init__.py:36: FutureWarning: Conversion of the second argument of issubdtype from `float` to `np.floating` is deprecated. In future, it will be treated as `np.float64 == np.dtype(float).type`.
  from ._conv import register_converters as _register_converters
Using TensorFlow backend.
Executing MHC Class I predictions
Converting .vcf to TSV
Completed
Splitting TSV into smaller chunks
Splitting TSV into smaller chunks - Entries 1-64
Completed
Generating Variant Peptide FASTA and Key Files
Generating Variant Peptide FASTA and Key Files - Entries 1-128
Completed
Processing entries for Allele HLA-A*02:01 and Epitope Length 8 - Entries 1-128
Running IEDB on Allele HLA-A*02:01 and Epitope Length 8 with Method NetMHCpan - Entries 1-128
Running IEDB on Allele HLA-A*02:01 and Epitope Length 8 with Method NetMHCpan - Entries 1-128 - Completed
Parsing IEDB Output for Allele HLA-A*02:01 and Epitope Length 8 - Entries 1-128
Parsing IEDB Output for Allele HLA-A*02:01 and Epitope Length 8 - Entries 1-128 - Completed
Processing entries for Allele HLA-A*02:01 and Epitope Length 9 - Entries 1-128
Running IEDB on Allele HLA-A*02:01 and Epitope Length 9 with Method NetMHCpan - Entries 1-128
Running IEDB on Allele HLA-A*02:01 and Epitope Length 9 with Method NetMHCpan - Entries 1-128 - Completed
Parsing IEDB Output for Allele HLA-A*02:01 and Epitope Length 9 - Entries 1-128
Parsing IEDB Output for Allele HLA-A*02:01 and Epitope Length 9 - Entries 1-128 - Completed
Processing entries for Allele HLA-A*02:01 and Epitope Length 10 - Entries 1-128
Running IEDB on Allele HLA-A*02:01 and Epitope Length 10 with Method NetMHCpan - Entries 1-128
Running IEDB on Allele HLA-A*02:01 and Epitope Length 10 with Method NetMHCpan - Entries 1-128 - Completed
Parsing IEDB Output for Allele HLA-A*02:01 and Epitope Length 10 - Entries 1-128
Parsing IEDB Output for Allele HLA-A*02:01 and Epitope Length 10 - Entries 1-128 - Completed
Processing entries for Allele HLA-A*02:01 and Epitope Length 11 - Entries 1-128
Running IEDB on Allele HLA-A*02:01 and Epitope Length 11 with Method NetMHCpan - Entries 1-128
Running IEDB on Allele HLA-A*02:01 and Epitope Length 11 with Method NetMHCpan - Entries 1-128 - Completed
Parsing IEDB Output for Allele HLA-A*02:01 and Epitope Length 11 - Entries 1-128
Parsing IEDB Output for Allele HLA-A*02:01 and Epitope Length 11 - Entries 1-128 - Completed
Processing entries for Allele HLA-B*35:01 and Epitope Length 8 - Entries 1-128
Running IEDB on Allele HLA-B*35:01 and Epitope Length 8 with Method NetMHCpan - Entries 1-128
Running IEDB on Allele HLA-B*35:01 and Epitope Length 8 with Method NetMHCpan - Entries 1-128 - Completed
Parsing IEDB Output for Allele HLA-B*35:01 and Epitope Length 8 - Entries 1-128
Parsing IEDB Output for Allele HLA-B*35:01 and Epitope Length 8 - Entries 1-128 - Completed
Processing entries for Allele HLA-B*35:01 and Epitope Length 9 - Entries 1-128
Running IEDB on Allele HLA-B*35:01 and Epitope Length 9 with Method NetMHCpan - Entries 1-128
Running IEDB on Allele HLA-B*35:01 and Epitope Length 9 with Method NetMHCpan - Entries 1-128 - Completed
Parsing IEDB Output for Allele HLA-B*35:01 and Epitope Length 9 - Entries 1-128
Parsing IEDB Output for Allele HLA-B*35:01 and Epitope Length 9 - Entries 1-128 - Completed
Processing entries for Allele HLA-B*35:01 and Epitope Length 10 - Entries 1-128
Running IEDB on Allele HLA-B*35:01 and Epitope Length 10 with Method NetMHCpan - Entries 1-128
Running IEDB on Allele HLA-B*35:01 and Epitope Length 10 with Method NetMHCpan - Entries 1-128 - Completed
Parsing IEDB Output for Allele HLA-B*35:01 and Epitope Length 10 - Entries 1-128
Parsing IEDB Output for Allele HLA-B*35:01 and Epitope Length 10 - Entries 1-128 - Completed
Processing entries for Allele HLA-B*35:01 and Epitope Length 11 - Entries 1-128
Running IEDB on Allele HLA-B*35:01 and Epitope Length 11 with Method NetMHCpan - Entries 1-128
Running IEDB on Allele HLA-B*35:01 and Epitope Length 11 with Method NetMHCpan - Entries 1-128 - Completed
Parsing IEDB Output for Allele HLA-B*35:01 and Epitope Length 11 - Entries 1-128
Parsing IEDB Output for Allele HLA-B*35:01 and Epitope Length 11 - Entries 1-128 - Completed
Combining Parsed IEDB Output Files
Completed
Running Binding Filters
Completed
Running Coverage Filters
Completed
Running Transcript Support Level Filter
Traceback (most recent call last):
  File "/home/xx/anaconda3/bin/pvacseq", line 11, in <module>
    sys.exit(main())
  File "/home/xx/anaconda3/lib/python3.6/site-packages/tools/pvacseq/main.py", line 99, in main
    args[0].func.main(args[1])
  File "/home/xx/anaconda3/lib/python3.6/site-packages/tools/pvacseq/run.py", line 171, in main
    pipeline.execute()
  File "/home/xx/anaconda3/lib/python3.6/site-packages/lib/pipeline.py", line 344, in execute
    PostProcessor(**post_processing_params).execute()
  File "/home/xx/anaconda3/lib/python3.6/site-packages/lib/post_processor.py", line 28, in execute
    self.execute_transcript_support_level_filter()
  File "/home/xx/anaconda3/lib/python3.6/site-packages/lib/post_processor.py", line 77, in execute_transcript_support_level_filter
    ['Transcript Support Level'],
  File "/home/xx/anaconda3/lib/python3.6/site-packages/lib/filter.py", line 25, in execute
    data = data.query(expression)
  File "/home/xx/anaconda3/lib/python3.6/site-packages/pandas/core/frame.py", line 2844, in query
    res = self.eval(expr, **kwargs)
  File "/home/xx/anaconda3/lib/python3.6/site-packages/pandas/core/frame.py", line 2959, in eval
    return _eval(expr, inplace=inplace, **kwargs)
  File "/home/xx/anaconda3/lib/python3.6/site-packages/pandas/core/computation/eval.py", line 296, in eval
    ret = eng_inst.evaluate()
  File "/home/xx/anaconda3/lib/python3.6/site-packages/pandas/core/computation/engines.py", line 76, in evaluate
    res = self._evaluate()
  File "/home/xx/anaconda3/lib/python3.6/site-packages/pandas/core/computation/engines.py", line 123, in _evaluate
    return ne.evaluate(s, local_dict=scope, truediv=truediv)
  File "/home/xx/anaconda3/lib/python3.6/site-packages/numexpr/necompiler.py", line 802, in evaluate
    _names_cache[expr_key] = getExprNames(ex, context)
  File "/home/xx/anaconda3/lib/python3.6/site-packages/numexpr/necompiler.py", line 709, in getExprNames
    ex = stringToExpression(text, {}, context)
  File "/home/xx/anaconda3/lib/python3.6/site-packages/numexpr/necompiler.py", line 299, in stringToExpression
    ex = eval(c, names)
  File "<expr>", line 1, in <module>
  File "/home/xx/anaconda3/lib/python3.6/site-packages/numexpr/expressions.py", line 96, in func
    raise TypeError("unsupported object type: %s" % type(x))
TypeError: unsupported object type: <class 'NoneType'>
```

**Comment by susannasiebert** [comment]
This seems like a bug in the transcript level filter. Could you please provide me with your input VCF so I can replicate the failure?

**Comment by qxiong2** [comment]
I use the input VCF in pvacseq_example_data
It is available at here https://pvactools.readthedocs.io/en/latest/pvacseq/getting_started.html
Thanks!

**Comment by susannasiebert** [comment]
Unfortunately, I'm unable to reproduce this error locally. A couple of questions:
- What version of pvactools do you have installed?
- Could you please attach your input VCF? I just want to double check that I'm using the same one as you are.
- Could you please execute the following command and post the output from it: `pip freeze`

**Comment by qxiong2** [comment]
I installed the latest pvactools (1.3.0)
Sorry, I don't know how to attached a file here.
When I ran 'pip freeze', I got the following
absl-py==0.7.0
alabaster==0.7.10
anaconda-client==1.6.14
anaconda-navigator==1.8.7
anaconda-project==0.8.2
appdirs==1.4.3
argh==0.26.2
asn1crypto==0.24.0
astor==0.7.1
astroid==1.6.3
astropy==3.0.2
attrs==18.1.0
Babel==2.5.3
backcall==0.1.0
backports.shutil-get-terminal-size==1.0.0
beautifulsoup4==4.6.0
biopython==1.73
bitarray==0.8.1
bkcharts==0.2
blaze==0.11.3
bleach==1.5.0
bokeh==0.13.0
boto==2.48.0
Bottleneck==1.2.1
certifi==2018.4.16
cffi==1.11.5
chardet==3.0.4
click==6.7
clickclick==1.2.2
cloudpickle==0.5.3
clyent==1.2.2
colorama==0.3.9
conda==4.6.3
conda-build==3.10.5
conda-verify==2.0.0
connexion==1.4.2
contextlib2==0.5.5
cryptography==2.2.2
cycler==0.10.0
Cython==0.28.2
cytoolz==0.9.0.1
dask==0.17.5
datashape==0.5.4
decorator==4.3.0
distributed==1.21.8
docutils==0.14
entrypoints==0.2.3
et-xmlfile==1.0.1
fastcache==1.0.2
filelock==3.0.4
Flask==1.0.2
Flask-Cors==3.0.4
gast==0.2.2
gevent==1.3.0
glob2==0.6
gmpy2==2.0.8
greenlet==0.4.13
grpcio==1.18.0
h5py==2.7.1
heapdict==1.0.0
html5lib==0.9999999
idna==2.6
imageio==2.3.0
imagesize==1.0.0
inflection==0.3.1
ipykernel==4.8.2
ipython==6.4.0
ipython-genutils==0.2.0
ipywidgets==7.2.1
isort==4.3.4
itsdangerous==0.24
jdcal==1.4
jedi==0.12.0
Jinja2==2.10
jsonschema==2.6.0
jupyter==1.0.0
jupyter-client==5.2.3
jupyter-console==5.2.0
jupyter-core==4.4.0
jupyterlab==0.32.1
jupyterlab-launcher==0.10.5
Keras==2.2.4
Keras-Applications==1.0.7
Keras-Preprocessing==1.0.9
kiwisolver==1.0.1
lazy-object-proxy==1.3.1
llvmlite==0.23.1
locket==0.2.0
lxml==4.2.1
Markdown==3.0.1
MarkupSafe==1.0
matplotlib==2.2.2
mccabe==0.6.1
mhcflurry==1.2.2
mhcnames==0.4.8
mhcnuggets==2.2
mistune==0.8.3
mkl-fft==1.0.0
mkl-random==1.0.1
more-itertools==4.1.0
mpmath==1.0.0
msgpack-python==0.5.6
multipledispatch==0.5.0
navigator-updater==0.2.1
nbconvert==5.3.1
nbformat==4.4.0
networkx==2.1
nltk==3.3
nose==1.3.7
notebook==5.5.0
numba==0.38.0
numexpr==2.6.5
numpy==1.14.3
numpydoc==0.8.0
odo==0.5.1
olefile==0.45.1
openpyxl==2.5.3
packaging==17.1
pandas==0.23.0
pandocfilters==1.4.2
parso==0.2.0
partd==0.3.8
path.py==11.0.1
pathlib2==2.3.2
pathtools==0.1.2
patsy==0.5.0
pep8==1.7.1
pexpect==4.5.0
pickleshare==0.7.4
Pillow==5.1.0
pkginfo==1.4.2
pluggy==0.6.0
ply==3.11
prompt-toolkit==1.0.15
protobuf==3.6.1
psutil==5.4.5
ptyprocess==0.5.2
pvactools==1.3.0
py==1.5.3
py-postgresql==1.2.1
pycodestyle==2.4.0
pycosat==0.6.3
pycparser==2.18
pycrypto==2.6.1
pycurl==7.43.0.1
pyflakes==1.6.0
Pygments==2.2.0
pylint==1.8.4
pymp-pypi==0.4.2
pyodbc==4.0.23
pyOpenSSL==18.0.0
pyparsing==2.2.0
pysam==0.15.2
PySocks==1.6.8
pytest==3.5.1
pytest-arraydiff==0.2
pytest-astropy==0.3.0
pytest-doctestplus==0.1.3
pytest-openfiles==0.3.0
pytest-remotedata==0.2.1
python-dateutil==2.7.3
pytz==2018.4
PyVCF==0.6.8
PyWavelets==0.5.2
PyYAML==3.12
pyzmq==17.0.0
QtAwesome==0.4.4
qtconsole==4.3.1
QtPy==1.4.1
requests==2.18.4
rope==0.10.7
ruamel-yaml==0.15.35
scikit-image==0.13.1
scikit-learn==0.19.1
scipy==1.1.0
seaborn==0.8.1
Send2Trash==1.5.0
simanneal==0.4.2
simplegeneric==0.8.1
singledispatch==3.4.0.3
six==1.11.0
snowballstemmer==1.2.1
sortedcollections==0.6.1
sortedcontainers==1.5.10
Sphinx==1.7.4
sphinxcontrib-websupport==1.0.1
spyder==3.2.8
SQLAlchemy==1.2.7
statsmodels==0.9.0
swagger-spec-validator==2.1.0
sympy==1.1.1
tables==3.4.3
tblib==1.3.2
tensorboard==1.9.0
tensorflow==1.9.0
termcolor==1.1.0
terminado==0.8.1
testpath==0.3.1
toolz==0.9.0
tornado==5.0.2
tqdm==4.31.1
traitlets==4.3.2
typing==3.6.4
unicodecsv==0.14.1
urllib3==1.22
watchdog==0.9.0
wcwidth==0.1.7
webencodings==0.5.1
Werkzeug==0.14.1
wget==3.2
widgetsnbextension==3.2.1
wrapt==1.10.11
xlrd==1.1.0
XlsxWriter==1.0.4
xlwt==1.3.0
zict==0.1.3
You are using pip version 10.0.1, however version 19.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

**Comment by qxiong2** [comment]
Hi Susanna,


I installed the latest pvactools (1.3.0)

See attached for input.vcf

When I ran 'pip freeze', I got the following
absl-py==0.7.0
alabaster==0.7.10
anaconda-client==1.6.14
anaconda-navigator==1.8.7
anaconda-project==0.8.2
appdirs==1.4.3
argh==0.26.2
asn1crypto==0.24.0
astor==0.7.1
astroid==1.6.3
astropy==3.0.2
attrs==18.1.0
Babel==2.5.3
backcall==0.1.0
backports.shutil-get-terminal-size==1.0.0
beautifulsoup4==4.6.0
biopython==1.73
bitarray==0.8.1
bkcharts==0.2
blaze==0.11.3
bleach==1.5.0
bokeh==0.13.0
boto==2.48.0
Bottleneck==1.2.1
certifi==2018.4.16
cffi==1.11.5
chardet==3.0.4
click==6.7
clickclick==1.2.2
cloudpickle==0.5.3
clyent==1.2.2
colorama==0.3.9
conda==4.6.3
conda-build==3.10.5
conda-verify==2.0.0
connexion==1.4.2
contextlib2==0.5.5
cryptography==2.2.2
cycler==0.10.0
Cython==0.28.2
cytoolz==0.9.0.1
dask==0.17.5
datashape==0.5.4
decorator==4.3.0
distributed==1.21.8
docutils==0.14
entrypoints==0.2.3
et-xmlfile==1.0.1
fastcache==1.0.2
filelock==3.0.4
Flask==1.0.2
Flask-Cors==3.0.4
gast==0.2.2
gevent==1.3.0
glob2==0.6
gmpy2==2.0.8
greenlet==0.4.13
grpcio==1.18.0
h5py==2.7.1
heapdict==1.0.0
html5lib==0.9999999
idna==2.6
imageio==2.3.0
imagesize==1.0.0
inflection==0.3.1
ipykernel==4.8.2
ipython==6.4.0
ipython-genutils==0.2.0
ipywidgets==7.2.1
isort==4.3.4
itsdangerous==0.24
jdcal==1.4
jedi==0.12.0
Jinja2==2.10
jsonschema==2.6.0
jupyter==1.0.0
jupyter-client==5.2.3
jupyter-console==5.2.0
jupyter-core==4.4.0
jupyterlab==0.32.1
jupyterlab-launcher==0.10.5
Keras==2.2.4
Keras-Applications==1.0.7
Keras-Preprocessing==1.0.9
kiwisolver==1.0.1
lazy-object-proxy==1.3.1
llvmlite==0.23.1
locket==0.2.0
lxml==4.2.1
Markdown==3.0.1
MarkupSafe==1.0
matplotlib==2.2.2
mccabe==0.6.1
mhcflurry==1.2.2
mhcnames==0.4.8
mhcnuggets==2.2
mistune==0.8.3
mkl-fft==1.0.0
mkl-random==1.0.1
more-itertools==4.1.0
mpmath==1.0.0
msgpack-python==0.5.6
multipledispatch==0.5.0
navigator-updater==0.2.1
nbconvert==5.3.1
nbformat==4.4.0
networkx==2.1
nltk==3.3
nose==1.3.7
notebook==5.5.0
numba==0.38.0
numexpr==2.6.5
numpy==1.14.3
numpydoc==0.8.0
odo==0.5.1
olefile==0.45.1
openpyxl==2.5.3
packaging==17.1
pandas==0.23.0
pandocfilters==1.4.2
parso==0.2.0
partd==0.3.8
path.py==11.0.1
pathlib2==2.3.2
pathtools==0.1.2
patsy==0.5.0
pep8==1.7.1
pexpect==4.5.0
pickleshare==0.7.4
Pillow==5.1.0
pkginfo==1.4.2
pluggy==0.6.0
ply==3.11
prompt-toolkit==1.0.15
protobuf==3.6.1
psutil==5.4.5
ptyprocess==0.5.2
pvactools==1.3.0
py==1.5.3
py-postgresql==1.2.1
pycodestyle==2.4.0
pycosat==0.6.3
pycparser==2.18
pycrypto==2.6.1
pycurl==7.43.0.1
pyflakes==1.6.0
Pygments==2.2.0
pylint==1.8.4
pymp-pypi==0.4.2
pyodbc==4.0.23
pyOpenSSL==18.0.0
pyparsing==2.2.0
pysam==0.15.2
PySocks==1.6.8
pytest==3.5.1
pytest-arraydiff==0.2
pytest-astropy==0.3.0
pytest-doctestplus==0.1.3
pytest-openfiles==0.3.0
pytest-remotedata==0.2.1
python-dateutil==2.7.3
pytz==2018.4
PyVCF==0.6.8
PyWavelets==0.5.2
PyYAML==3.12
pyzmq==17.0.0
QtAwesome==0.4.4
qtconsole==4.3.1
QtPy==1.4.1
requests==2.18.4
rope==0.10.7
ruamel-yaml==0.15.35
scikit-image==0.13.1
scikit-learn==0.19.1
scipy==1.1.0
seaborn==0.8.1
Send2Trash==1.5.0
simanneal==0.4.2
simplegeneric==0.8.1
singledispatch==3.4.0.3
six==1.11.0
snowballstemmer==1.2.1
sortedcollections==0.6.1
sortedcontainers==1.5.10
Sphinx==1.7.4
sphinxcontrib-websupport==1.0.1
spyder==3.2.8
SQLAlchemy==1.2.7
statsmodels==0.9.0
swagger-spec-validator==2.1.0
sympy==1.1.1
tables==3.4.3
tblib==1.3.2
tensorboard==1.9.0
tensorflow==1.9.0
termcolor==1.1.0
terminado==0.8.1
testpath==0.3.1
toolz==0.9.0
tornado==5.0.2
tqdm==4.31.1
traitlets==4.3.2
typing==3.6.4
unicodecsv==0.14.1
urllib3==1.22
watchdog==0.9.0
wcwidth==0.1.7
webencodings==0.5.1
Werkzeug==0.14.1
wget==3.2
widgetsnbextension==3.2.1
wrapt==1.10.11
xlrd==1.1.0
XlsxWriter==1.0.4
xlwt==1.3.0
zict==0.1.3
You are using pip version 10.0.1, however version 19.0.2 is available.
You should consider upgrading via the 'pip install --upgrade pip' command.

Thanks,
Qing

 
 
 
-----Original Message-----

**Comment by tdw1221** [comment]
I'm actually having the same problem as well in some of my samples, and I'd be happy to provide my vcf files and code if needed, thank you!

**Comment by susannasiebert** [comment]
This issue has been fixed in version 1.3.1. Please upgrade to the latest version and let me know if that works for you.
